src/Routing/Router.php

```diff
@@ -1074,6 +1075,26 @@ public static function routes()
     }
 
     /**
+     * Get a MiddlewareQueue of middleware that matches the provided path.
+     *
+     * @param string $path The URL path to match for.
+     * @return \Cake\Http\MiddlewareQueue|null Either a queue or null if there are no matching middleware.
+     */
+    public static function getMatchingMiddleware($path)
+    {
+        if (!static::$initialized) {
+            static::_loadRoutes();
+        }
+
+        $middleware = static::$_collection->getMatchingMiddleware($path);
+        if ($middleware) {
```

Comment:
Small thing: If we invert, the happy case would be the final line in the method, aborting early with a return null then instead.